test/integration/short/load-balancing-tests.js

```diff
@@ -144,9 +144,69 @@ context('with a reusable 3 node cluster', function () {
         });
       }, helper.finish(client, done));
     });
+    it('should target the correct replica using user-provided Buffer routingKey', function (done) {
+      // Use [0] which should map to node 1
+      testWithQueryOptions((client) => ({
+        routingKey: Buffer.from([0])
+      }), '1', done);
+    });
+    it('should target the correct replica using user-provided Token routingKey', function (done) {
+      testWithQueryOptions((client) => {
+        // Find TokenRange for host 2 and use the end token (which is inclusive).
+        const host = helper.findHost(client, 2);
+        const ranges = client.metadata.getTokenRangesForHost('ks_network_rp1', host);
+        const range = ranges.values().next().value;
+        return { routingKey: range.end };
+      }, '2', done);
+    });
+    it('should target the correct replica using user-provided TokenRange routingKey', function (done) {
+      testWithQueryOptions((client) => {
+        // Find TokenRange for host 3 and use it.
+        const host = helper.findHost(client, 3);
+        const ranges = client.metadata.getTokenRangesForHost('ks_network_rp1', host);
+        const range = ranges.values().next().value;
+        return { routingKey: range };
+      }, '3', done);
+    });
+    it('should throw TypeError if invalid routingKey type is provided', function () {
+      const client = new Client({
+        policies: { loadBalancing: new TokenAwarePolicy(new RoundRobinPolicy()) },
+        keyspace: 'ks_network_rp1',
+        contactPoints: helper.baseOptions.contactPoints
+      });
+      const query = 'select * from system.local';
+      return client.execute(query, [], { routingKey: 'this is not valid' })
```

Comment:
I think we need to use the callback-based interface here, as using promises will catch errors that are unexpectedly thrown along with rejected promises.

We should test that `client.execute()` will not sync throw an error
```
// client.execute() call should not throw
client.execute(query, [], { routingKey: 'this is not valid' }, err => {
  helper.assertInstanceOf(err, TypeError);
  done();
});
```